### ISSUE TITLE
Virtual merge commit related "patch does not apply"

### ISSUE THREAD

**Issue by distorhead** [werf/werf#3349]
Inconsistent state of docker image detected:

```
# docker inspect docker.pkg.github.com/werf/werf/werfio-ng:b391986a9d16ed6edb0424b6adbfb69f4d433418b093d3f414359019-1615558918822
...
            "Labels": {
                "maintainer": "Jordon Bedwell <jordon@envygeeks.io>",
                "werf": "werfio-ng",
                "werf-cache-version": "1.2",
                "werf-docker-image-name": "a8bf1a2c-78be-42cc-ac93-bb9cf1d42224",
                "werf-git-2eee5f2c8f7511b270624a3b0085da9552e97aa5bfc85ab13404f6d4891cdec0-commit": "e3632eecbbf5e60808cf17bb31a751df5abce6ee",
                "werf-git-2eee5f2c8f7511b270624a3b0085da9552e97aa5bfc85ab13404f6d4891cdec0-type": "directory",
                "werf-git-2eee5f2c8f7511b270624a3b0085da9552e97aa5bfc85ab13404f6d4891cdec0-virtual-merge": "true",
                "werf-git-2eee5f2c8f7511b270624a3b0085da9552e97aa5bfc85ab13404f6d4891cdec0-virtual-merge-from-commit": "a49387630cd895629a521c19b0b7a14ef4b8a3cc",
                "werf-git-2eee5f2c8f7511b270624a3b0085da9552e97aa5bfc85ab13404f6d4891cdec0-virtual-merge-into-commit": "1657954a8e9a5137cc3c2c47f21aacce46a85143",
                "werf-image": "false",
                "werf-project-repo-commit": "e3632eecbbf5e60808cf17bb31a751df5abce6ee",
                "werf-stage-content-digest": "78fc2bfa60619209b29e990c902a71a41336ee1348ba854c17a49434",
                "werf-stage-digest": "b391986a9d16ed6edb0424b6adbfb69f4d433418b093d3f414359019",
                "werf-version": "v1.2.9+fix8"
            }
...
```

This says that commit `e3632eecbbf5e60808cf17bb31a751df5abce6ee` is a virtual merge from `a49387630cd895629a521c19b0b7a14ef4b8a3cc` to `1657954a8e9a5137cc3c2c47f21aacce46a85143`.

But when inspecting commit `e3632eecbbf5e60808cf17bb31a751df5abce6ee` in the git worktree:

```
# git show e3632eecbbf5e60808cf17bb31a751df5abce6ee
commit e3632eecbbf5e60808cf17bb31a751df5abce6ee
Merge: 5a16d9796 a49387630
Author: valent-ex <valent.rus@gmail.com>
Date:   Fri Mar 12 14:09:56 2021 +0000

    Merge a49387630cd895629a521c19b0b7a14ef4b8a3cc into 5a16d9796465a6c6b363a9360db3b55162932eb9
```

This says that commit `e3632eecbbf5e60808cf17bb31a751df5abce6ee` is merge from `a49387630cd895629a521c19b0b7a14ef4b8a3cc` to the different commit `5a16d9796465a6c6b363a9360db3b55162932eb9`.

More info:
 - `5a16d9796465a6c6b363a9360db3b55162932eb9` is the most latest commit in the master at the moment;
 - `1657954a8e9a5137cc3c2c47f21aacce46a85143` was the most latest commit in the master some time ago.

**Comment by distorhead** [comment]
Inconsistent image was built in this pipeline: https://github.com/werf/werf/runs/2095715805?check_suite_focus=true

**Comment by distorhead** [comment]
Fixed in the https://github.com/werf/werf/pull/4075.
Fix available since v1.2.57: https://github.com/werf/werf/blob/main/CHANGELOG.md#1257-2022-01-19.
